### PR TITLE
Add resource based configuration for IOpipe. (Fixes #170, Fixes #169)

### LINKED ISSUE #169: Generic Entry Point handler should have a fallback with system properties
Currently the generic entry point only looks at an environment variable to set the entry point, it should have a condition to check from a system property.

### LINKED ISSUE #170: Allow for IOpipe configuration to be stored in `/application.properties` in the JAR
This will allow a secondary system to allow for properties which are stored within the JAR, it could be used as part of a build process in situations where one is already used and is the preferred method without needing to actually set any environment variables. It would only be used as a fallback (environment variables, system properties, then `/application.properties`).

### PR DESCRIPTION
**Pull request by XerTheSquirrel**
This will allow for using resources to set IOpipe properties which will have to be included at compile time.

### CODE REVIEW

**Comment by XerTheSquirrel** [comment]
Fixes #169 as well.

**Comment by XerTheSquirrel** [comment]
This is ready for review!

File: README.md [hunk 1]
```diff
@@ -225,6 +225,13 @@ environment variables.
    * Alternative URL for the collector, this is mostly used for debugging and
      experimentation with newer collectors.
 
+Alternatively a configuration may be specified in the root of the JAR with a
+standard properties format (`key=value`) which is named `iopipe.properties`.
+The configuration values are only used if they have not been specified by
+system properties or environment variables. Generally using this is not
```
Comment (pselle) [inline README.md]: If this is not recommended, why offer the instructions?

Comment (XerTheSquirrel) [reply README.md]: I have it there to just say that it is available for use, otherwise nobody would know about it.

**Pull request merged**